bin/generate_data.js

```diff
@@ -4,15 +4,26 @@ const dataSource = require('../minecraft-data/data/dataPaths')
 const fs = require('fs')
 const path = require('path')
 
-const data = 'module.exports =\n{\n' + Object
+const data = 'const { pathToFileURL } = require(\'url\')\nmodule.exports =\n{\n' + Object
   .keys(dataSource)
   .map(k1 =>
     "  '" + k1 + "': {\n" + Object
       .keys(dataSource[k1])
       .map(k2 =>
         "    '" + k2 + "': {" + '\n' + Object
           .keys(dataSource[k1][k2])
-          .map(k3 => "      '" + k3 + "': require('./minecraft-data/data/" + dataSource[k1][k2][k3] + '/' + k3 + ".json')")
+          .map(k3 => {
+            const loc = `minecraft-data/data/${dataSource[k1][k2][k3]}/`
+            try {
+              // Check if the file can be loaded as JSON
+              require('../' + loc + k3 + '.json')
+              return `      get ${k3} () { return require("./${loc}${k3}.json") }`
```

Comment:
The getter will break webpack

Reply:
I tested with webpack earlier (no custom rules) and just tested prismarine-web-client with it and it seems to work.

![image](https://user-images.githubusercontent.com/13713600/129515731-952a1e0e-8047-4e61-8aa7-2e5941d9f91b.png)

in data.js (I think this is missing a publish script to build data.js since I had to manually build it)
```js
  'pc': {
    '0.30c': {
      get blocks () { return require("./minecraft-data/data/pc/0.30c/blocks.json") },
      get protocol () { return require("./minecraft-data/data/pc/0.30c/protocol.json") },
      get version () { return require("./minecraft-data/data/pc/0.30c/version.json") }
    },
```

My understanding is it just parses the AST for `require` calls instead of actually evaluating it.

Reply:
That's surprising to me as I expected webpack to bundle all these files as one
Does it still work if you run npm run build to get the built thing and run the http server on it ?
(Instead of the dev server, is that what you ran ?)

Reply:
Oops, yeah I ran the dev server. But it still seems to work on prod server. I would post the index.js file but it's too big for gist so an excerpt after minification:
```js
hantments(){return a(30841)},get recipes(){return a(9379)},get instruments(){return a(46583)},get materials(){return a(30365)},get language(){return a(76037)},get entities(){return a(2403)},get protocol(){return a(70404)},get windows(){return a(29663)},get version(){return a(58506)},get foods(){return a(80430)},get particles(){return a(86342)},get blockLoot(){return a(29482)},get entityLoot(){return a(15142)},get loginPacket(){return a(56006)},get tints(){return a(69207)},get mapIcons(){return a(15107)}},"1.17.1":{get blocks(){return a(46423)},get blockCollisionShapes(){return a(79877)},get biomes(){return a(87175)},get effects(){return a(65552)},get items(){return a(70573)},get enchantments(){return a(30841)},get recipes(){return a(9379)},get instruments(){return a(46583)},get materials(){return a(30365)},get language(){return a(76037)},get entities(){return a(2403)},get protocol(){return a(20272)},get windows(){return a(29663)},get version(){return a(34441)},get foods(){return a(80430)},get particles(){return a(86342)},get blockLoot(){return a(29482)},get entityLoot(){return a(15142)},get loginPacket(){return a(56006)},get tints(){return a(69207)},get mapIcons(){return a(15107)}}},bedrock:{.14:{get blocks(){return a(5984)},get biomes(){return a(85990)},get items(){return a(87286)},get protocol(){return a(47096)},get version(){return a(52808)}},.15:{get blocks(){return a(7571)},get biomes(){return a(85990)},get items(){return a(73200)},get protocol(){return a(73406)},get version(){return a(77073)}},"1.0":{get blocks(){return a(17092)},get biomes(){return a(85990)},get items(){return a(58732)},get version(){return a(43072)}},"1.16.201":{get protocol(){return a(71727)}},"1.16.210":{get protocol(){return a(9598)}},"1.16.220":{get blockStates(){return a(77786)},get steve(){return a(22367)},get steveGeometry(){return a(52931)},get protocol(){return a(24801)}},"1.17.0":{get blockStates(){return a(49941)},get blockCollisionShapes(){return a(90670)},get biomes(){return a(19083)},get instruments(){return a(5105)},get protocol(){return a(98952)},get windows(){return a(62497)},get steve(){return a(22367)},get steveGeometry(){return a(52931)},get blocksB2J(){return a(17896)},get blocksJ2B(){return a(24251)}},"1.17.10":{get blocks(){return a(42582)},get blockStates(){return a(68626)},get blockCollisionShapes(){return a(42187)},get biomes(){return a(19083)},get items(){return a(71266)},get recipes(){return a(26678)},get instruments(){return a(5105)},get protocol(){return a(85571)},get windows(){return a(62497)},get blocksB2J(){return a(42778)},get blocksJ2B(){return a(32066)},get steve(){return a(22367)},get steveGeometry(){return a(52931)}}}}},64784:(e,t,a)=>{const n=a(487),o=a(62913),l={pc:a(95181),bedrock:a(26055)},i={},r={},s={},c={};function _(e,t,a){const n=i[e];for(const e in n){const t=n[e];n[t.majorVersion]=n[t.majorVersion]||t}const o=n[t]?.dataVersion??0,l=t=>{throw new RangeError(`Version '${t}' not found in [${Object.keys(n).join(" ; ")}] for ${e}`)};return this[">="]=e=>n[e]?o>=n[e].dataVersion:l(e),this[">"]=e=>n[e]?o>n[e].dataVersion:l(e),this["<"]=e=>n[e]?o<n[e].dataVersion:l(e),this["<="]=e=>n[e]?o<=n[e].dataVersion:l(e),this["=="]=e=>n[e]?o===n[e].dataVersion:l(e),this.type=e,this.majorVersion=a,this}"undefined"!=typeof window&&alert("yo I work!"),["pc","bedrock"].forEach((function(e){for(let t=0;t<l[e].length;t++)l[e][t].dataVersion||(l[e][t].dataVersion=-l[e].length+t);i[e]=o.buildIndexFromArray(l[e],"minecraftVersion"),r[e]=o.buildIndexFromArray(l[e].slice().reverse(),"majorVersion"),s[e]=o.buildIndexFromArrayNonUnique(l[e].filter((function(e){return!e.usesNetty})),"version"),c[e]=o.buildIndexFromArrayNonUnique(l[e].filter((function(e){return e.usesNetty})),"version")}));const p={};function d(e,t,a){const n=(e+"").split("_"),o=a||(2===n.length?n[0]:"pc"),l=2===n.length?n[1]:e;let p;if(m[o][l])p=l;else if(i[o][l])p=i[o][l].majorVersion;else{if(t&&s[o][l])return d(s[o][l][0].minecraftV
```

![image](https://user-images.githubusercontent.com/13713600/129545633-cc4d3aa1-3beb-4a38-a34e-66d0c0fedee2.png)

the [worker.js file](https://gist.githubusercontent.com/extremeheat/a2df347daaa8e243d694c4feea34f13d/raw/058f579e4fbd0464dfa242bf0cefaad93388c80a/worker.js) also has the json

Reply:
ok that's cool

I think it will make browserify fail (used in minecraft data doc), but I guess we could consider switching to webpack there